src/main/java/org/jabref/gui/desktop/JabRefDesktop.java

```diff
@@ -168,7 +169,38 @@ private static void openExternalFilePlatformIndependent(Optional<ExternalFileTyp
      * @throws IOException
      */
     public static void openFolderAndSelectFile(Path fileLink) throws IOException {
-        NATIVE_DESKTOP.openFolderAndSelectFile(fileLink);
+        if (Objects.isNull(fileLink)) {
+            return;
+        }
+        boolean usingDefault = Globals.prefs.getBoolean(JabRefPreferences.USE_DEFAULT_FILE_BROWSER_APPLICATION);
+
+        if(usingDefault) {
+            NATIVE_DESKTOP.openFolderAndSelectFile(fileLink);
+        }else{
+            String absolutePath = fileLink.toAbsolutePath().getParent().toString();
+            String command = Globals.prefs.get(JabRefPreferences.FILE_BROWSER_COMMAND);
+            if (!command.isEmpty()) {
+                command = command.replaceAll("\\s+", " "); // normalize white spaces
+
+                // replace the placeholder if used
+                command = command.replace("%DIR", absolutePath);
+                String[] subcommands = command.split(" ");
+
+                JabRefGUI.getMainFrame().output(Localization.lang("Executing command \"%0\"...", command));
```

Comment:
I think `output` is deprecated, you should instead use `DialogService`.

Reply:
ok, I'll change this

Reply:
JabRefDesktop does not have field DialogService field, so should I add a class level filed Or a local variable for DialogService instance?

Reply:
`JabRefGUI.getMainFrame().getDialogService()` should work.

Reply:
You should pass the DialogService as parameter to the method. You may need to do this at several locations.

Reply:
Ok thanks, I've replaced that line by this JabRefGUI.getMainFrame().getDialogService().notify()
in my local machine.